### PR TITLE
POL-189 Credential Service changes

### PR DESCRIPTION
**Pull request by KTebrewal**
Added the credential Service changes and a required check for input parameter - Tag Key.

### Description

Update listed Azure policy to support the credential service. The changes include-
-Replace auth declaration with the credential declaration.
-Remove long_description field and use new info hash for Version, providers and service.
-Update readme to replace credential instructions with instructions to use credential service.
-Update CWF definitions as needed.

### Issues Resolved

Added new credential service

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### Running template Link
**Azure: Tag Resources with Resource Group Name**
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e4cdf9bcbdfbd000119bdb8
image

### CODE REVIEW

**Comment by KTebrewal** [comment]
> Added the credential Service changes and a required check for input parameter - Tag Key.
> 
> ### Description
> Update listed Azure policy to support the credential service. The changes include-
> -Replace auth declaration with the credential declaration.
> -Remove long_description field and use new info hash for Version, providers and service.
> -Update readme to replace credential instructions with instructions to use credential service.
> -Update CWF definitions as needed.
> 
> ### Issues Resolved
> Added new credential service
> 
> ### Contribution Check List
> * [x]  All tests pass.
> * [x]  New functionality includes testing.
> * [x]  New functionality has been documented in the README if applicable
> * [x]  New functionality has been documented in CHANGELOG.MD
> 
> ### Running template Link

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt [hunk 1]
```diff
@@ -153,7 +164,7 @@ end
 
 policy "policy_azure_resource" do
   validate_each $ds_filtered_resources do
-    summary_template "{{ len data }} Azure Resources out of compliance. Azure Subscription ID: {{parameters.param_azure_sub}}"
+    summary_template "{{ len data }} Azure Resources out of compliance."
     detail_template <<-EOS
 # Azure Resources Incorrectly Tagged
 | Resource Name | Resource Group | Azure Region | Resource ID | Tag Value |
```
Comment (cdwilhelm) [inline compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: Please add subscription name to the result and sort by subscription name and resource name

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt [hunk 2]
```diff
@@ -185,30 +195,31 @@ escalation "tag_resources" do
...
-
-  call get_access_token($param_azure_tenant_id) retrieve $access_token
-
+define tag_resources($data, $param_tag_key) return $all_responses do
+  $$log = []
+  $all_responses = []
+  $$subscriptionId = ""; 
+  foreach $item in $data do
+    $$subscriptionId = $item["subscriptionId"]
+  end
```
Comment (cdwilhelm) [inline compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: This block doesn't look right. shouldn't the end statement happen after the tagging is completed.  Why are you setting a global variable.

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt (hunk 1 above)
Comment (KTebrewal) [reply compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: @cdwilhelm I have added the subscription name and sorted as well.

File: compliance/tags/azure_rg_tags/azure_resource_group_tags.pt (hunk 2 above)
Comment (KTebrewal) [reply compliance/tags/azure_rg_tags/azure_resource_group_tags.pt]: I have updated the logic as per your comments. Works fine.

**Pull request merged**